common/persistence/cassandra/cassandraHistoryV2Persistence.go

```diff
@@ -193,6 +196,10 @@ func (h *cassandraHistoryV2Persistence) ReadHistoryBranch(request *p.InternalRea
 		}
 	}
 
+	if !found && len(request.NextPageToken) == 0 {
```

Comment:
It's better to do this check in HistoryV2Store instead of Cassandra/SQL implementation. 
In some cases fork succeeds but haven't append events yet, we probably don't want to return error here.

Reply:
I agree with you that certralizing logic is better. But in this case, we have two components (Cassandra/SQL) implementing the same interface. I think it's kind of weird if the two implementations have different behaviors and the difference is handled by a higher layer.

Reply:
For the fork case you mentioned, I think the request.NextPageToken will not be empty? So no error will be returned?

Reply:
Doing it in HistoryV2Store will let both Cassandra/SQL have the same behavior without writing code in two places. That's why we have this higher layer.
I am not sure why you think difference if we write this code in the higher layer.

The only difference I know is historyV1 have it in lower layer, but essentially we can/should also do it in the higher layer.

For that case, request.NextPageToken is nil, check this line: https://github.com/uber/cadence/blob/master/common/persistence/historyV2Store.go#L372

Reply:
Oh, sorry, I was actually think about V1 where the behavior is different for two implementations. And yes, for V2, right now both SQL and Cassandra will return empty response if there's no row and we can do the check in higher layer to see if an error is needed. I agree and will update.